### PR TITLE
Add buildkite Build step

### PR DESCRIPTION
**Pull request by gogochan**
## Change Summary

This PR adds a dev pipeline that does following:
- [x] Buildkite build step - handling build and git diff test
- [x] Buildkite check step - handling static and pipeline test
- [x] Adding make target `elastic-package` to use in the pipeline.
- [x] Configure Buildkite to cancel intermediate build

Here's a run of this pipeline: https://buildkite.com/elastic/endpoint-package/builds/36

Github status does not block any PR yet. This can be enabled later.

I am leaning toward creating a separated PR for publishing `main` or other releasable branches, and will not integrate this using pr-bot.

image

### Sample values

Sample document:

```json


```

### For mapping changes:

- [ ] I ran `make` after making the schema changes, and committed all changes
- [ ] If these field(s) are "exception"-able, I made a companion PR to Kibana adding it (see [Readme](https://github.com/elastic/endpoint-package#exceptionable))
- [ ] If this is a `metadata` change, I also updated both transform destination schemas to match

### For Transform changes:

- [ ] The new transform successfully starts in Kibana
- [ ] The corresponding transform destination schema was updated if necessary

### CODE REVIEW

File: .buildkite/pipeline.yml.py [hunk 1]
```diff
@@ -1,14 +1,83 @@
...
-        ]
-    }))
+    steps = [
+        {
+            "label": "Build",
+            "command": "make",
+            "key": "build",
+            #"agents": test_agent,
+            "artifact_paths": [
+                "build/packages/*.zip",
```
Comment (pzl) [inline .buildkite/pipeline.yml.py:20]: does this do anything? What's the side-effect of marking something an `artifact_paths`?

File: .buildkite/pipeline.yml.py [hunk 2]
```diff
@@ -1,14 +1,83 @@
...
+            #"agents": test_agent,
+            "notify": [
+                {
+                    "github_commit_status": {
+                        "context": "Check Pipeline Test",
+                    },
+                },
+            ],
+        },
+        {
```
Comment (pzl) [inline .buildkite/pipeline.yml.py:58]: These will both require that a stack is running.  And the environment variables are set to point to whatever stack is use

**Review by pzl — COMMENTED** [review]
Will buildkite execute this change as part of its run?

**Comment by gogochan** [comment]
This is not complete yet, I'm pushing it up so that I can see validate the workflow.
Do not review at the moment.

File: .buildkite/pipeline.yml.py (hunk 1 above)
Comment (gogochan) [reply .buildkite/pipeline.yml.py:27]: if there's anything under build/packages/*.zip, it stashes it. This behavior is similar to https://github.com/elastic/endpoint-package/blob/main/.ci/Jenkinsfile#L63C1-L64C1

**Comment by gogochan** [comment]
This is now ready for a review.

**Review by pzl — APPROVED** [review]
lets goooooo

**Review by dwhyrock — APPROVED** [review]

**Comment by gogochan** [comment]
https://github.com/elastic/security-team/issues/6786
for tracking purpose

**Pull request merged**